packages/flutter/test/material/flexible_space_bar_test.dart

```diff
@@ -470,6 +470,57 @@ void main() {
     );
   });
 
+  testWidgets('FlexibleSpaceBar sets width constraints for the title - override expandedScale', (WidgetTester tester) async {
+    const double titleFontSize = 20.0;
+    const double height = 300.0;
+    late double width;
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: Builder(
+            builder: (BuildContext context) {
+              width = MediaQuery.of(context).size.width;
+              return CustomScrollView(
+                slivers: <Widget>[
+                  SliverAppBar(
+                    expandedHeight: height,
+                    pinned: true,
+                    stretch: true,
+                    flexibleSpace: FlexibleSpaceBar(
+                      expandedScale: 3,
+                      titlePadding: EdgeInsets.zero,
+                      title: Text(
+                        'X' * 2000,
+                        maxLines: 1,
+                        overflow: TextOverflow.ellipsis,
+                        style: const TextStyle(fontSize: titleFontSize),
+                      ),
+                      centerTitle: false,
+                    ),
+                  ),
+                ],
+              );
+            },
+          ),
+        ),
+      ),
+    );
+
+    // The title is scaled and transformed to be 3 times bigger, when the
```

Comment:
It would be a little safer to measure the title's bounds before and after it has been expanded (rather than assuming that its width will be the same as `MediaQuery.of(context).size.width;`